packages/clay-form/src/Input.tsx

```diff
@@ -0,0 +1,63 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import classNames from 'classnames';
+import React, {useMemo} from 'react';
+
+export interface IProps extends React.InputHTMLAttributes<HTMLInputElement> {
+	/**
+	 * Gets the accepted characters of the input
+	 * element values
+	 */
+	allowedCharacters?: RegExp;
```

Comment:
Is this type of functionality really needed for our use case? We could likely just leave this to the user by having them manage their own `value` and `onChange`. I may be missing something here though...

Reply:
I ended up bringing this in when we introduced MultiSelect to v2, that was difficult there, now with low-level I really do not see if there's any more need, but I think it would be interesting to consider some features and within Clay if we delegate always for developers to deal with this, with low-level they may have to deal with all logic always this can generate overload for those who consume.

Reply:
Yeah I agree. I think we can keep the low-level simple and then anything we find our users re-implementing and creating, we can then create a high-level component for ease of use.

Reply:
Yeah, I'm going to remove this functionality for now and depending on usage we'll add a new API or high-level if need to.